remotespark/livyclientlib/sparkcontroller.py

```diff
@@ -52,56 +42,53 @@ def get_all_sessions_endpoint_info(self, connection_string):
         return [str(s) for s in sessions]
 
     def cleanup(self):
-        self.client_manager.clean_up_all()
+        self.session_manager.clean_up_all()
 
     def cleanup_endpoint(self, connection_string):
         for session in self.get_all_sessions_endpoint(connection_string):
             session.delete()
 
     def delete_session_by_name(self, name):
-        self.client_manager.delete_client(name)
+        self.session_manager.delete_client(name)
 
     def delete_session_by_id(self, connection_string, session_id):
         http_client = self._http_client_from_connection_string(connection_string)
-        r = http_client.get("/sessions/{}".format(session_id), [200, 404])
-        if r.status_code != 404:
-            session = self._create_livy_session(connection_string, {"kind": r.json()["kind"]},
-                                                self.ipython_display, session_id, False)
+        try:
+            r = http_client.get_session(session_id)
```

Comment:
Variable names should be meaningful, and I can't find `r` meaningful variable name. I thouhgt it's `r_kernal` so I suggested this name. [I think this is not part of this PR but since we are re-organizing this'll be good as well]